### PR TITLE
Bring analytics back online 

### PR DESCRIPTION
**Pull request by nathanleclaire**
still need to test, but this is looking decent 

requesting review but not quite merge yet

### CODE REVIEW

File: src/renderer/data/Config/configState.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { stat } from 'fs';
```
Comment (SvenDowideit) [inline src/renderer/data/Config/configState.ts:2]: I presume this is unused? as we're on the client side, so shouldn't be accessing the fs?

Comment (nathanleclaire) [reply src/renderer/data/Config/configState.ts:2]: good point, this must be VS code gone bad

File: src/renderer/common/analytics.ts [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
+      }
+    } else {
+      // eslint-disable-next-line no-console
+      console.log('analytics event', event);
     }
   }
 };
 
-// Analytics('openApp', {});  // TODO: disabled this one as it crashes sometimes
+analytics('openApp', {}); // TODO: disabled this one as it crashes sometimes
```
Comment (SvenDowideit) [inline src/renderer/common/analytics.ts:28]: I guess a bit of testing to make sure the TODO is removable :)

File: src/renderer/data/accounts/getAccount.ts [hunk 3]
```diff
@@ -94,8 +94,8 @@ export const renderData = (account: AccountInfo | undefined) => {
 };
 
 // TODO: this should look up a persistent key: string map
-export const getHumanName = (key: AccountInfo | sol.PublicKey) => {
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const getHumanName = (_key: AccountInfo | sol.PublicKey) => {
   /* eslint-disable no-console */
```
Comment (SvenDowideit) [inline src/renderer/data/accounts/getAccount.ts:99]: OI ! this one too :p

File: src/renderer/common/analytics.ts (hunk 2 above)
Comment (nathanleclaire) [reply src/renderer/common/analytics.ts:28]: do you remember if it was a hooks related warning? it didn't like any `analytics` call until I took that `useAppSelector` out of it

File: src/renderer/data/accounts/getAccount.ts (hunk 3 above)
Comment (nathanleclaire) [reply src/renderer/data/accounts/getAccount.ts:99]: this one what. this isn't really related to the pr tbh but it was spamming the console something terrible

**Comment by SvenDowideit** [comment]
Biggest question is - can all the electron specific code on the renderer be separated out to other files to make it easier to re-use this repo for a web build too?

File: src/renderer/common/analytics.ts (hunk 2 above)
Comment (SvenDowideit) [reply src/renderer/common/analytics.ts:28]: I am going to say - its plausible

File: src/renderer/data/accounts/getAccount.ts (hunk 3 above)
Comment (SvenDowideit) [reply src/renderer/data/accounts/getAccount.ts:99]: yup - its just that in removing the console.log, you can thus also remove the eslint-disable :)

we need a lint instruction linter-er

**Comment by nathanleclaire** [comment]
> Biggest question is - can all the electron specific code on the renderer be separated out to other files to make it easier to re-use this repo for a web build too?

It definitely should, but let's defer that to another day while we get v0.3.0 out

A big Q about that is how we have non-deterministic actions in the `useEffect` hooks. 

probably something like this? (with `effect` being a prop)

Screen Shot 2022-04-21 at 2 01 00 PM

**Comment by nathanleclaire** [comment]
One thing to note on config in particular is that we can't really do that with localStorage imo. User's settings will need to be persisted for upgrades (I assume localstorage will reset in the Electron app each time at least). Also, FWIW on the web due to adblocks I consider it fair game to get set analytics enabled to true by default and skip the prompt.

**Comment by SvenDowideit** [comment]
yeah, i'm not trying to say "use localstorage only", i'm wanting a logical code partition so that local storage is used as a client buffer, and then we sync from there to whatever makes sense for that deployment.

**Review by SvenDowideit — APPROVED** [review]
LGTM

**Comment by nathanleclaire** [comment]
> yeah, i'm not trying to say "use localstorage only", i'm wanting a logical code partition so that local storage is used as a client buffer, and then we sync from there to whatever makes sense for that deployment.
oh, right -- fair enough!

**Pull request merged**